lib/Infer/AliveDriver.cpp

```diff
@@ -260,6 +315,68 @@ souper::AliveDriver::synthesizeConstants(souper::Inst *RHS) {
   return synthesizeConstantUsingSolver(t, Consts);
 }
 
+std::map<souper::Inst *, llvm::APInt>
+souper::AliveDriver::synthesizeConstantsWithCegis(souper::Inst *RHS, InstContext &IC) {
+  std::map<souper::Inst *, llvm::APInt> ConstMap;
+
+  std::map<std::string, Inst *> Consts;
+  std::set<Inst *> Visited;
+  getReservedConsts(RHS, Consts, Visited);
+  assert(!Consts.empty());
+
+  smt::expr TriedExpr(true);
+
+  RExprCache.clear();
+  IR::Function RHSF;
+  if (!translateRoot(RHS, nullptr, RHSF, RExprCache)) {
+    if (DebugLevel > 2)
+      llvm::errs() << "Failed to translate RHS.\n";
+    // TODO: Eventually turn this into an assertion
+    return {};
+  }
+
+  tools::Transform t;
+  t.tgt = std::move(RHSF);
+
+  unsigned Tried = 0;
+  while (true) {
+    // First Query
+    {
+      ReturnLHSRAII foo{t, LHSF};
+      t.src = std::move(LHSF);
+
+      Tried ++;
+      // exceeds MAX_TRIES, stop immediately
+      if (Tried > MaxTries) {
+        if (DebugLevel > 2)
+          llvm::errs()<<"Time of tries reached maximum\n";
+        return {};
+      }
+
+      ConstMap = performCegisFirstQuery(t, Consts, TriedExpr);
+
+      if (ConstMap.empty())
```

Comment:
what does it mean when this happens?

Reply:
No more guesses produced, terminate immediately.

Reply:
I mean: under what circumstances does this happen?

Reply:
This happens when the first query is unsat, see the implementation of performCegisFirstQuery

Reply:
ok-- I think this deserves a printout at some debug level so we can see this happening